open-media-match/pyproject.toml

```diff
@@ -19,6 +19,15 @@ dependencies = [
     "flask",
 ]
 
+[project.optional-dependencies]
```

Comment:
Lol, @dougneal , who I think removed the all alias in a previous version.

To toss my hat in the ring:
1. I think it's helpful to separate test and production dependencies
2. Where I had landed on with Doug is that it would be fine to just have installing anything be the default
3. I am okay with having the test/all split brought back, but defer to @dougneal  for additional opinons.

Reply:
I would definitely expect the dev mode dependencies to be different to the production mode dependencies, so this is expected in some form or another. 

The `pyproject.toml` way of doing things has a built-in concept of [build-time dependencies](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/#build-time-dependencies).  This means these tools are listed under the `requires` section of the `build-system` stanza (above). This makes it more explicit that "these deps are for dev/build time" whereas putting them under optional dependencies is a bit more ambiguous in its intent.

The `pyproject.toml` can also contain configuration stanzas for each of these tools, see [this example](https://github.com/pamelafox/flask-db-quiz-example/blob/main/pyproject.toml).

Reply:
but wouldn't adding them all to the `build-system` mean it will always be installed? we wouldn't want them installed on the prod container tho. unless you mean to create one for prod that overrides the dependencies with a smaller subset?

Reply:
@dougneal I'm not sure I am able to follow how to get this working without the use of optional dependencies. I'm having a hard time understanding how to get the dependencies to be part of the build-system such as they are installed on `--editable` 😓 .